src/sentry/seer/explorer/tools.py

```
@@ -151,21 +92,70 @@ def execute_trace_query_table(
             "trace",
         ]
 
+    return execute_table_query(
+        org_id=org_id,
+        dataset="spans",
+        fields=fields,
+        query=query,
+        sort=sort,
+        per_page=per_page,
+        stats_period=stats_period,
+        project_ids=project_ids,
+    )
+
+
+def execute_table_query(
+    *,
+    org_id: int,
+    dataset: str,
+    fields: list[str],
+    query: str,
+    sort: str,
+    per_page: int,
+    stats_period: str,
+    project_ids: list[int] | None = None,
+    project_slugs: list[str] | None = None,
+    sampling_mode: SAMPLING_MODES = "NORMAL",
+) -> dict[str, Any] | None:
+    """
+    Execute a query to get table data by calling the events endpoint.
+
+    Arg notes:
+        project_ids: The IDs of the projects to query. Cannot be provided with project_slugs.
+        project_slugs: The slugs of the projects to query. Cannot be provided with project_ids.
+        If neither project_ids nor project_slugs are provided, all active projects will be queried.
+    """
+    try:
+        organization = Organization.objects.get(id=org_id)
+    except Organization.DoesNotExist:
+        logger.warning("Organization not found", extra={"org_id": org_id})
+        return None
+
+    if not fields:
+        # Must pass in at least one field.
```

should we raise an error here or add a log?

---

I think i'll remove this check and defer to the endpoint's behavior, if it errors it'll be logged, if it defaults then we'll use that default. We shouldn't be passing this from seer though.